xinference/core/restful_api.py

```diff
@@ -266,11 +274,51 @@ def serve(self):
             methods=["POST"],
             response_model=ChatCompletion,
         )
-        app.include_router(self._router)
-        app = gr.mount_gradio_app(app, self._gradio_block, path="/")
+
+        self._router.add_api_route(
+            "/v1/gradio/{model_uid}", self.build_interface, methods=["POST"]
+        )
+
+        self._app.include_router(self._router)
+
+        class SPAStaticFiles(StaticFiles):
+            async def get_response(self, path: str, scope):
+                response = await super().get_response(path, scope)
+                if response.status_code == 404:
+                    response = await super().get_response(".", scope)
+                return response
+
+        try:
+            lib_location = os.path.abspath(
+                os.path.dirname(__import__("xinference").__file__)
+            )
+            ui_location = os.path.join(lib_location, "web/ui/build/")
+        except ImportError as e:
+            raise ImportError(f"Xinference is imported incorrectly: {e}")
+
+        if os.path.exists(ui_location):
+
+            @self._app.get("/")
+            def read_main():
+                response = RedirectResponse(url="/ui/")
+                return response
+
+            self._app.mount(
+                "/ui/",
+                SPAStaticFiles(directory=ui_location, html=True),
+                name="Xinference",
```

Comment:
What does `name` do?

Reply:
I believe `name` is supposed to give the webpage a name or header. In this case, it did not work due to the redirection from the main page. I have removed this line.